graph/src/data/subgraph/mod.rs

```diff
@@ -666,6 +673,95 @@ impl UnresolvedMapping {
     }
 }
 
+#[derive(Clone, PartialEq, Debug)]
+pub struct UnifiedMappingApiVersion(Option<Version>);
+
+impl UnifiedMappingApiVersion {
+    pub fn equal_or_greater_than(&self, other_version: &'static Version) -> bool {
+        assert!(
+            other_version >= &API_VERSION_0_0_5,
+            "api versions before 0.0.5 should not be used for comparison"
+        );
+        match &self.0 {
+            Some(version) => version >= other_version,
+            None => false,
+        }
+    }
+
+    pub fn try_from_versions<'a>(
+        versions: impl Iterator<Item = &'a Version>,
+    ) -> Result<Self, DifferentMappingApiVersions> {
+        let unique_versions: BTreeSet<Version> = versions.into_iter().cloned().collect();
+
+        let all_below_referential_version = unique_versions.iter().all(|v| *v < API_VERSION_0_0_5);
+        let all_the_same = unique_versions.len() == 1;
+
+        let unified_version: Option<Version> = match (all_below_referential_version, all_the_same) {
+            (false, false) => return Err(unique_versions.into()),
+            (false, true) => Some(unique_versions.iter().nth(0).unwrap().deref().clone()),
+            (true, _) => None,
+        };
+
+        Ok(UnifiedMappingApiVersion(unified_version))
+    }
+}
+
+#[test]
+fn unified_mapping_api_version_from_iterator() {
```

Comment:
I see you tried to keep the test close to the tested code, which makes sense but I think we usually just keep tests at the bottom of the file.